app/src/main/java/com/example/secret/model/PostsModel.java

```diff
@@ -36,6 +53,40 @@ public void getPost(String postId, Listener<Post> successListener, Listener<Void
         firebaseModel.getPost(postId, modelSuccessListener, failListener);
     }
 
+    public LiveData<List<Post>> getAllPosts() {
+        if (postList == null) {
+            postList = localDb.postDao().getAll();
+            refreshLatestPosts();
+        }
+        return postList;
+    }
+
+    public void refreshLatestPosts() {
+        EventPostsListLoadingState.setValue(PostsModel.LoadingState.LOADING);
+        // get local last update
+        Long localLastUpdate = Post.getLocalLastUpdate();
+        // get all updated recorde from firebase since local last update
+        firebaseModel.getAllPostsSince(localLastUpdate, list -> executor.execute(() -> {
```

Comment:
There is no need for ThreadPoolExecutor. see `addOnCompleteListener`, `addOnSuccessListener`, `addOnFailureListener`

Reply:
That's Eliav's code.